docs/spec/spec.md

```diff
@@ -0,0 +1,1090 @@
+    * 3.2. [Initialization](#32-initialization)
+    * 3.3. [Functions](#33-functions)
+4. [Message receiver](#4-message-receiver)
+    * 4.1. [Configurations](#41-configurations)
+    * 4.2. [Initialization](#42-initialization)
+    * 4.3. [Functions](#43-functions)
+5. [Message listener](#5-message-listener)
+    * 5.1. [Configurations](#51-configurations)
+    * 5.2. [Initialization](#52-initialization)
+    * 5.3. [Functions](#53-functions)
+    * 5.4. [Caller](#54-caller)
+        * 5.4.1. [Functions](#541-functions)
+    * 5.5. [Validations](#55-validations)
+        * 5.5.1. [Attaching multiple services to a listener](#551-attaching-multiple-services-to-a-listener)
+        * 5.5.2. [Using `autoComplete` mode](#552-using-autocomplete-mode)
+    * 5.6. [Usage](#56-usage)
+6. [Administrator](#6-administrator)
+    * 6.1. [Initialization](#61-initialization)
+    * 6.2. [Functions](#62-functions)
+
+
+## 1. Overview
+
+Azure Service Bus is a highly reliable cloud messaging service from Microsoft that enables communication between 
+distributed applications through messages. It supports complex messaging features like FIFO messaging, publish/subscribe 
+models, and session handling, making it an ideal choice for enhancing application connectivity and scalability.
+
+This specification details the usage of Azure Service Bus in the context of queues and topics, enabling developers to build 
+robust distributed applications and microservices. These components facilitate the parallel, scalable, and fault-tolerant 
+handling of messages even in the face of network issues or service interruptions.
+
+Ballerina `asb` provides several core APIs:
+
+- `asb:MessageSender` - used to publish messages to Azure service bus queue or a topic.
+- `asb:MessageReceiver` - used to receive messages from an Azure service bus queue or a topic.
+- `asb:Listener` - used to asynchronously receive messages from an Azure service bus queue or topic.
+- `asb:Administrator` - used to perform administrative actions on an Azure service bus resource.
+
+## 2. Message
+
+An Azure Service Bus message is a unit of communication that carries data for exchange between components that interact with Microsoft's Azure service bus resource. Messages can encapsulate different types of data, including text, binary, and custom-defined objects structured as key-value pairs.
+
+A Message Batch in Azure Service Bus is a collection of messages that are sent as a single batched transaction to optimize performance and resource utilization. This approach is useful when sending multiple messages to the same queue or topic to ensure atomicity and reduce network calls.
+
+- `ApplicationProperties` record represents the application-specific properties.
+
+```ballerina
+public type ApplicationProperties record {|
+    # Key-value pairs for each brokered property (optional)
+    map<anydata> properties?;
+|};
+```
+
+- `Message` record represents the Azure service bus message.
```

Comment:
IMO better to swap the order

Reply:
Fixed with [e00aab7617ad94072eb328f0c7d76c234f61dabb](https://github.com/ballerina-platform/module-ballerinax-azure-service-bus/pull/228/commits/e00aab7617ad94072eb328f0c7d76c234f61dabb)